src/huggingface_hub/hf_api.py

```diff
@@ -1269,6 +1252,92 @@ def update_repo_visibility(
         r.raise_for_status()
         return r.json()
 
+    def rename_repo(
+        self,
+        from_id: str,
+        to_id: str,
+        repo_type: Optional[str] = None,
+        token: Optional[str] = None,
+    ):
+        """
+        Rename a repository from namespace1/repo_name1 to namespace2/repo_name2
+
+        Renaming can be used for three use cases
+        - Renaming a repository within same user.
+        - Renaming a repository within same organization.
+        - Move repository from user to an organization.
```

Comment:
are you sure we cannot move repository from organization to (currently loggedin) user? 

cc @SBrandeis @Pierrci @coyotte508 for the latest on what's planned/shipped

Reply:
> are you sure we cannot move repository from organization to (currently loggedin) user?

It's not possible at the moment, unless the one doing the move is an hf user (which may be changed by https://github.com/huggingface/moon-landing/pull/2134)

Reply:
(other cases can be tracked in https://github.com/huggingface/moon-landing/issues/1451)